control/preimage/src/main.rs

```diff
@@ -71,70 +151,115 @@ fn parse_hex_bytes(v: &str) -> Result<Bytes, String> {
+}
 
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, ValueEnum)]
 pub enum Format {
     Hex,
     Binary,
 }
 
-#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Debug, ValueEnum)]
-pub enum GatewayOperatingModeArg {
-    Normal,
-    RejectingOutboundMessages,
-}
-
 struct StaticConfig<'a> {
     api: &'a str,
+    asset_hub_api: &'a str,
 }
 
 struct Context {
     api: Box<OnlineClient<PolkadotConfig>>,
+    asset_hub_api: Box<OnlineClient<PolkadotConfig>>,
 }
 
+#[cfg(feature = "rococo-local")]
+static CONFIG: StaticConfig<'static> = StaticConfig {
+    api: "ws://127.0.0.1:11144",
+    asset_hub_api: "ws://127.0.0.1:12144",
+};
+
 #[cfg(feature = "rococo")]
 static CONFIG: StaticConfig<'static> = StaticConfig {
     api: "wss://rococo-bridge-hub-rpc.polkadot.io",
+    asset_hub_api: "wss://rococo-asset-hub-rpc.polkadot.io",
 };
 
 #[cfg(feature = "kusama")]
 static CONFIG: StaticConfig<'static> = StaticConfig {
     api: "wss://kusama-bridge-hub-rpc.polkadot.io",
+    asset_hub_api: "wss://kusama-asset-hub-rpc.polkadot.io",
 };
 
 #[cfg(feature = "polkadot")]
 static CONFIG: StaticConfig<'static> = StaticConfig {
     api: "wss://polkadot-bridge-hub-rpc.polkadot.io",
+    asset_hub_api: "wss://polkadot-asset-hub-rpc.polkadot.io",
 };
 
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
+async fn main() {
+    if let Err(err) = run().await {
+        eprintln!("{err}");
+    }
+}
+
+async fn run() -> Result<(), Box<dyn std::error::Error>> {
     let cli = Cli::parse();
 
 	let api: OnlineClient<PolkadotConfig> = OnlineClient::from_url(CONFIG.api)
 		.await
 		.expect("can not connect to bridgehub");
 
+    let asset_hub_api: OnlineClient<PolkadotConfig> = OnlineClient::from_url(CONFIG.asset_hub_api)
+    .await
+    .expect("can not connect to assethub");
+
+
     let context = Context {
-        api: Box::new(api)
+        api: Box::new(api),
+        asset_hub_api: Box::new(asset_hub_api)
     };
 
     let call = match &cli.command {
-        Command::GatewayOperatingMode { mode } => {
-            commands::gateway_operating_mode(&context, *mode).await?
+        Command::ForceCheckpoint(params) => {
+            let call = commands::force_checkpoint(params);
+            send_xcm_bridge_hub(&context, vec![call]).await?
+        },
+        Command::Initialize(params) => {
+            let (set_pricing_parameters, set_ethereum_fee) = commands::pricing_parameters(&context, &params.pricing_parameters).await?;
+            let call1 = send_xcm_bridge_hub(&context, vec![
+                commands::force_checkpoint(&params.force_checkpoint),
+                commands::gateway_operating_mode(&params.gateway_operating_mode),
+                commands::set_gateway_address(&params.gateway_address),
+                set_pricing_parameters,
+            ]).await?;
+            let call2 = send_xcm_asset_hub(&context, vec![set_ethereum_fee]).await?;
+            utility_batch(vec![call1, call2])
```

Comment:
IIUC we also need to set the xcm version of Ethereum in AH or the `reserve_transfer_assets` will fail. 

Codes for that in https://github.com/Snowfork/polkadot-sdk/blob/5da7df7a823a88b3ff93bdc612b6e2006f30bb96/cumulus/parachains/integration-tests/emulated/tests/bridges/bridge-hub-rococo/src/tests/snowbridge.rs#L380-L383 and local setup in
https://github.com/Snowfork/snowbridge/blob/cc6f1236088dfba23bec54f8a7882d218f261a62/web/packages/test/scripts/configure-substrate.sh#L82-L85

Reply:
Is this just for local E2E? As AssetHub is already running in Polkadot though for many years, and presumably already has XCM configured.

Reply:
Not sure but it did require an entry exist in storage `supported_versions`, just checked `asset-hub` on Rococo testnet
<img width="1122" alt="image" src="https://github.com/Snowfork/snowbridge/assets/4383920/b4bcfd4b-d7e2-44d9-92a7-07efa05b611c">

Reply:
I had to set this in Rococo for the bridge to work. This is not a generic xcm configuration information, this sets the XCM version for our bridge location specifically. We need to add it to the preimage.